### PR TITLE
po/remove auto hack - exposure & basecurve

### PR DESCRIPTION
**Pull request by TurboGit**
Remove specific handling for exposure & basecurve.
    
The default and auto-apply is now using standard reload_defaults().
   
This also ensure that when the reset button is clicked for those
modules the proper default is set (the same as if the history
stack is reset).
    
At the same time this will now be properly handled by the new
auto-init support for presets/style/copy&paste support introduced
in #13492, #13450, #13432.
    
Also we don't require a reboot anymore when the auto-apply per-camera
basecurve setting is changed.

### CODE REVIEW

**Comment by jenshannoschwalm** [comment]
Works good for me :-)

**Comment by TurboGit** [comment]
@wpferguson @victoryforce @kmilos : Any idea to why this is failing on Windows? Missing symbols for the regex support!!!

**Comment by victoryforce** [comment]
> @wpferguson @victoryforce @kmilos : Any idea to why this is failing on Windows? Missing symbols for the regex support!!!

regex.h is a POSIX specific header file. To use the corresponding functions under Windows, we will have to find a portable implementation library. I am aware of two options: libgnurx and libsystre.

**Comment by TurboGit** [comment]
> regex.h is a POSIX specific header file. To use the corresponding functions under Windows, we will have to find a portable implementation library. I am aware of two options: libgnurx and libsystre.

Well I better be using the GLib support regex. I'll give it a try.

File: .ci/ci-script-windows.sh [hunk 1]
```diff
@@ -82,7 +82,7 @@ cat > "$FONTCONFIG_FILE" <<EOF
 EOF
 
 execute 'Installing base-devel and toolchain'  pacman -S --needed --noconfirm mingw-w64-x86_64-{toolchain,clang,cmake}
-execute 'Installing dependencies' pacman -S --needed --noconfirm  mingw-w64-x86_64-{exiv2,lcms2,dbus-glib,openexr,sqlite3,libxslt,libsoup,libavif,libwebp,libsecret,lua,graphicsmagick,openjpeg2,gtk3,pugixml,libexif,osm-gps-map,libgphoto2,flickcurl,drmingw,gettext,python3,iso-codes,lensfun}
+execute 'Installing dependencies' pacman -S --needed --noconfirm  mingw-w64-x86_64-{exiv2,lcms2,dbus-glib,openexr,sqlite3,libxslt,libsoup,libavif,libwebp,libsecret,lua,graphicsmagick,openjpeg2,gtk3,pugixml,libexif,osm-gps-map,libgphoto2,flickcurl,drmingw,gettext,python3,iso-codes,lensfun,libsystre}
```
Comment (kmilos) [inline .ci/ci-script-windows.sh:85]: Hm, this is not where we do it, should be in .fithub/workflows/CI.yaml (and nightly.yaml) - and I have a PR in mind deleting this useles file anyway...

**Comment by kmilos** [comment]
Yeah, I don't like this solution very much...

File: .ci/ci-script-windows.sh (hunk 1 above)
Comment (TurboGit) [reply .ci/ci-script-windows.sh]: This is now removed anyway since I've a new implementation based on GLib.

**Comment by TurboGit** [comment]
> And I don't see where the regex came in? Sql query?

Yes, we are matching the basecure profile against some camera maker/model. That's old code supporting the legacy display referred workflow.

**Comment by kmilos** [comment]
There could be an alternative query, I saw a few places where SQL queries are different `#ifdef _WIN32`...

**Comment by TurboGit** [comment]
> There could be an alternative way to query, I saw a few places where SQL queries are different `#ifdef _WIN32`?

I'm not sure to understand. Here we are using the same list of definition (added here since long time and method deprecated as we don't use new camera specific base curve) to create the presets (in SQL db) with filters based on maker/model and to create the actual default for the module based on the same maker/model data but in C.

**Comment by kmilos** [comment]
> I'm not sure to understand.

I meant stuff like [this.](https://github.com/darktable-org/darktable/blob/4bdae2c20b9b3a6439fc041ff50cae034cd6dfa7/src/common/film.c#L94-L102) But then again I don't know SQL so it might be unrelated to regex...

**Comment by TurboGit** [comment]
> I meant stuff like [this.](https://github.com/darktable-org/darktable/blob/4bdae2c20b9b3a6439fc041ff50cae034cd6dfa7/src/common/film.c#L94-L102) But then again I don't know SQL so it might be unrelated to regex...

Indeed, don't know why this #ifdef _WIN32 is there... No comment on the code. This will need to be checked on the Git history to find the commit. But anyway, this has no relation with regex here.

**Comment by kmilos** [comment]
> Indeed, don't know why this #ifdef _WIN32 is there...

Could be case sensitive vs case insensitive matching? Anyway, I'll try to take another look at the new changes later...

**Comment by TurboGit** [comment]
It is also important to note that this fix a bug introduced to support multiple- auto-preset of the same module (#12853). If you use "display-referred (legacy)" workflow and reset an history stack you'll end-up with multiple instances of basecurve. This is because the SQL query in develop.c to get the auto-preset was only returning a single preset per module. This required that the preset was created in the proper order for basecure as we want the more specific one (per camera basecurve) first. All this was not documented of course :) At least I found out about this bug and it is now fixed here. The good news is that no users using the dev version seems to be using the display-referred workflow or we would have quite some issues reported since a08bd88743abe8f719efa07d81a4b7978a0b7bf4 was merged in Jan 17th.

**Comment by TurboGit** [comment]
I have pushed an updated version which do case-insensitive & anchored match. This is safer.

@kmilos : If ok to you this is ready to be merged at this point.

**Comment by kmilos** [comment]
No more platform specific code, so ok with me 😉

**Comment by TurboGit** [comment]
Thanks!

**Pull request merged**